opal/mca/pmix/pmix114/pmix/examples/dynamic.c

```diff
@@ -45,12 +45,12 @@ int main(int argc, char **argv)
     uint32_t nprocs;
     char nsp2[PMIX_MAX_NSLEN+1];
     pmix_app_t *app;
-    char hostname[1024], dir[1024];
+    char hostname[MAXHOSTNAMELEN], dir[1024];
```

Comment:
Similar to libevent, the pmix component comes from outside Open MPI, too.  Patches for PMIx should likely go to their upstream, too.